src/data/SiteWise/ListAssociatedAssets.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  IoTSiteWiseClient,
+  paginateListAssociatedAssets,
+} from "@aws-sdk/client-iotsitewise";
+
+import type { AssetSummary } from "../../entities/SiteWise";
+import logger from "../../services/logger";
+
+class ListAssociatedAssets {
+  private client: IoTSiteWiseClient;
+
+  constructor(client: IoTSiteWiseClient) {
+    this.client = client;
+  }
+
+  public async invoke(
+    assetId: string,
+    hierarchyId: string,
+  ): Promise<AssetSummary[]> {
+    try {
+      const assetSummaries: AssetSummary[] = [];
+      for await (const page of paginateListAssociatedAssets(
```

Comment:
Would `Promise.all()` works better here? `Promise.all()` invokes all promises but for await only invokes one at a time. Does order matters here?